### PR TITLE
Adding Static Analysis

### PR DESCRIPTION
**Pull request by PavelApis**

### CODE REVIEW

File: src/main/java/my/minesweeper/papchelnikov/package-info.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/**
+ *
+ *
+ * @since 1.0
+ * @author papchelnikov
+ * @version 1.0
+ */
+
+package my.minesweeper.papchelnikov;
```
Comment (mfvanek) [inline src/main/java/my/minesweeper/papchelnikov/package-info.java:9]: You are on GitHub so your reverse domain name is **io.github.pavelapis**

File: Minesweeper.iml [hunk 2]
```diff
@@ -1,9 +1,13 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<module type="JAVA_MODULE" version="4">
-  <component name="NewModuleRootManager" inherit-compiler-output="true">
-    <exclude-output />
+<module org.jetbrains.idea.maven.project.MavenProjectsManager.isMavenModule="true" type="JAVA_MODULE" version="4">
```
Comment (mfvanek) [inline Minesweeper.iml:2]: Usually we don't add such files to the repository

File: config/checkstyle/checkstyle.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0"?>
+<!DOCTYPE module PUBLIC
+        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
+        "https://checkstyle.org/dtds/configuration_1_3.dtd">
+
+<module name="Checker">
+    <module name="JavadocPackage"/>
+    <module name="TreeWalker">
+        <module name="AvoidStarImport"/>
```
Comment (mfvanek) [inline config/checkstyle/checkstyle.xml:9]: Good for start but consider to add more rules

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    <artifactId>Minesweeper</artifactId>
+    <version>1.0-SNAPSHOT</version>
+
+    <properties>
+        <maven.compiler.source>17</maven.compiler.source>
+        <maven.compiler.target>17</maven.compiler.target>
+    </properties>
+
+    <build>
+    <pluginManagement>
```
Comment (mfvanek) [inline pom.xml:17]: Could you reformat this file?

File: pom.xml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+            </plugin>
+        </plugins>
+    </build>
+
+    <reporting>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-checkstyle-plugin</artifactId>
+                <version>3.1.2</version>
```
Comment (mfvanek) [inline pom.xml:61]: You shouldn't specify version here

File: src/main/java/my/minesweeper/papchelnikov/WelcomeFrame.java [hunk 6]
```diff
@@ -23,7 +32,7 @@ public WelcomeFrame(){
 
 
 
-        JButton start_game = new JButton("Start Game");
+        JButton start_game = new JButton("Start my.minesweeper.papchelnikov.Game");
```
Comment (mfvanek) [inline src/main/java/my/minesweeper/papchelnikov/WelcomeFrame.java:35]: In my view it's redundant changes

**Review by mfvanek — CHANGES_REQUESTED** [review]

File: config/checkstyle/checkstyle.xml (hunk 3 above)
Comment (PavelApis) [reply config/checkstyle/checkstyle.xml:22]: I has added more modules and refactored their warnings

File: pom.xml (hunk 4 above)
Comment (PavelApis) [reply pom.xml]: Done

File: src/main/java/my/minesweeper/papchelnikov/WelcomeFrame.java (hunk 6 above)
Comment (PavelApis) [reply src/main/java/my/minesweeper/papchelnikov/WelcomeFrame.java]: Yeep)

**Review by mfvanek — APPROVED** [review]
To my mind it's ready for merging

**Pull request merged**